src/main/java/org/nuxeo/build/ant/StudioExtraction.java

```diff
@@ -0,0 +1,169 @@
+    protected String extract = "*";
+
+    protected File todir;
+
+    protected String filename = "studio-registries.json";
+
+    protected boolean failOnEmpty = false;
+
+    protected Vector<ResourceCollection> rcs = new Vector<>();
+
+    /**
+     * Perform the extraction
+     *
+     * @throws BuildException if an error occurs.
+     */
+    @Override
+    public void execute() throws BuildException {
+        validateAttributes();
+        try {
+
+            new ContributionsExtractor(buildOptions()).publish();
+        } catch (IOException e) {
+            throw new BuildException(e);
+        }
+    }
+
+    public void validateAttributes() {
+        if (todir == null) {
+            throw new BuildException("todir attribute cannot be empty");
+        }
+
+        if (rcs.isEmpty()) {
+            throw new BuildException("Use nested FileSet(s) to set jar files to extract");
+        }
+    }
+
+    /**
+     * Add a set of files to extract from a nested element.
+     *
+     * @param fileSet a file set to extract.
+     */
+    public void addFileSet(FileSet fileSet) {
+        add(fileSet);
+    }
+
+    /**
+     * Add a resource collection of files to extract from a nested element.
+     *
+     * @param rc a resource collection to extract.
+     */
+    public void add(ResourceCollection rc) {
+        rcs.add(rc);
+    }
+
+    protected ExtractorOptions buildOptions() {
+        ExtractorOptions opts = new ExtractorOptions();
+        opts.setBuildDirectory(todir.getAbsolutePath());
+        opts.setExtract(getExtract());
+        opts.setFailOnEmpty(isFailOnEmpty());
+        opts.setOutput(getFilename());
+
+        List<String> files = new ArrayList<>();
+        for (ResourceCollection rc : rcs) {
+            if (rc instanceof FileSet) {
+                FileSet fs = (FileSet) rc;
+                File basedir = fs.getDirectoryScanner().getBasedir();
+                for (String file : fs.getDirectoryScanner().getIncludedFiles()) {
+                    files.add(new File(basedir, file).getAbsolutePath());
+                }
+            } else {
+                if (!rc.isFilesystemOnly()) {
+                    throw new BuildException(
+                            "Only FileSystem resources are supported.");
+                }
+
+                for (Resource r : rc) {
+                    if (!r.isExists()) {
+                        log("Warning: Could not find resource " + r.toLongString());
+                        continue;
+                    }
+
+                    final FileProvider fp = r.as(FileProvider.class);
+                    if (fp != null) {
+                        final FileResource fr = ResourceUtils.asFileResource(fp);
+                        files.add(fr.getFile().getAbsolutePath());
+                    }
+                }
+            }
+        }
+        opts.setJarFile(StringUtils.join(files, ","));
+        return opts;
+    }
+
+    public File getTodir() {
```

Comment:
No getter. Usually, Ant task do not need any getter, but setters for customizable fields, yes.
See https://ant.apache.org/manual/develop.html